### PR TITLE
IA-1437 disallow highcpu machine type for free credit projects

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-1437

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala:174]: Geraldine has some concerns about this; some workshops require `highcpu` clusters for legitimate purposes. It might be better to put specific disallowed machine types in leo.conf, which we could change as needed.

**Review by rtitle — CHANGES_REQUESTED** [review]
see comment above

Comms also asked to be informed about any changes around free credits projects, as it impacts workshops. So let's hold off on merging until we coordinate with them (it's also encouraging that we haven't had any new bitcoin users since yesterday).

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala
Comment (kyuksel) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala:84]: `Unallowed` instead of `Invalid` may be more correct

**Comment by Qi77Qi** [comment]
security team has turned on alerts, and will potentially shutdown services if they detect malicious behavior. Hence closing this PR.

**Pull request closed without merging**
